internal/cmd/process.go

```diff
@@ -457,32 +456,35 @@ func buildLogDisplayInput(executionId, pipelineExecutionId string) types.Pipelin
 func obtainRoutedInputsForExecution(routerUrl string, token string, stepKeys []string) (*primitive.RoutedInputListResponse, error) {
 	client := &http.Client{}
 
-	for i, step := range stepKeys {
-		stepKeys[i] = fmt.Sprintf("'%s'", step)
-	}
-	params := url.Values{}
-	params.Add("where", fmt.Sprintf("step_execution_id in (%s)", strings.Join(stepKeys, ",")))
-	listInputUrl := fmt.Sprintf("%s?%s", routerUrl, params.Encode())
+	inputItems := make([]primitive.RoutedInputResponse, 0)
+	for i := range stepKeys {
+		getInputUrl := fmt.Sprintf("%s/%s", routerUrl, stepKeys[i])
 
-	req, e := http.NewRequest(http.MethodGet, listInputUrl, nil)
-	if e != nil {
-		return nil, perr.InternalWithMessage("Failed to build request to obtain routed inputs.")
-	}
-	req.Header.Set("Authorization", "Bearer "+token)
-	resp, e := client.Do(req)
-	if e != nil {
-		return nil, perr.InternalWithMessage("Failed to obtain list of routed inputs.")
-	}
-	defer resp.Body.Close()
-	resBody, e := io.ReadAll(resp.Body)
-	if e != nil {
-		return nil, perr.InternalWithMessage("Failed reading response body.")
-	}
-	var inputs primitive.RoutedInputListResponse
-	e = json.Unmarshal(resBody, &inputs)
-	if e != nil {
-		return nil, perr.InternalWithMessage("Failed to deserialize the response body from input router.")
+		req, e := http.NewRequest(http.MethodGet, getInputUrl, nil)
+		if e != nil {
+			return nil, perr.InternalWithMessage("Failed to build request to obtain routed inputs.")
+		}
+
+		req.Header.Set("Authorization", "Bearer "+token)
+		resp, e := client.Do(req)
+		if e != nil {
+			return nil, perr.InternalWithMessage("Failed to obtain list of routed inputs.")
```

Comment:
Error message still references obtaining a list, should be amended to be singular?